portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,22 +20,29 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-import java.util.ArrayList;
+import java.util.HashMap;
+// import java.util.ArrayList;
 import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  ArrayList<String> messages = new ArrayList<>();
+  HashMap<String, String> messages = new HashMap<String, String>();
```

Comment:
1) Per my comment on the parent, can you use the interface here instead of the concrete type (Map instead of HashMap)?
2) Java can infer the types from the declaration.
3) Reduce visibility of member variables as much as possible.
```suggestion
  private final Map<String, String> messages = new HashMap<>();
```

Reply:
Ok, I will do this as well.